integration/tests/grid-tooltip.test.js

```diff
@@ -383,3 +385,39 @@ describe('tooltip', () => {
     });
   });
 });
+
+describe('tooltip in column renderer', () => {
+  let container, grid, cell, items;
+
+  beforeEach(async () => {
+    container = fixtureSync('<div></div>');
+    items = ['first', 'second', 'third'];
+    render(
+      html` <vaadin-grid .items="${items}">
+        <vaadin-grid-column
+          ${columnBodyRenderer(
+            (item) => html`
+              <div id="${item}">${item}</div>
+              <vaadin-tooltip for="${item}" .text="${item}"></vaadin-tooltip>
+            `,
+            items,
+          )}
+        ></vaadin-grid-column>
+      </vaadin-grid>`,
+      container,
+    );
+    await nextFrame();
+    grid = container.querySelector('vaadin-grid');
+    cell = getContainerCell(grid.$.items, 2, 0);
+  });
+
+  it('should still target correct element after grid content is sorted differently', async () => {
```

Comment:
This doesn't necessarily need an integration test because the bug can be produced without grid. Tried with the following test case in tooltip package:

```js
it('should still target correct element after sorting the items differently', async () => {
  const container = fixtureSync('<div></div>');
  function renderTooltips(items) {
    render(
      html`
        ${items.map(
          (item) => html`
            <vaadin-tooltip for="${item}"></vaadin-tooltip>
            <div id=${item}></div>
          `,
        )}
      `,
      container,
    );
  }

  renderTooltips(['bar', 'foo']);
  renderTooltips(['foo']);

  await nextFrame();
  expect(container.querySelector('vaadin-tooltip[for="foo"]').target).to.equal(container.querySelector('#foo'));
});
```

This test revealed an issue in `__setTargetById` where, due to the added asynchronizity, it might get invoked for elements that are no longer in the DOM, leading to an error. Could be fixed by adding this check:
```js
if (!this.isConnected) {
  return;
}
```